### DISCUSSION TITLE
Always cleaning before building certain AUR packages

### DISCUSSION THREAD

**Discussion by zsolt-donca** [Morganamilo/paru#419] [Q&A]
Is there a way to configure paru to always clean the build directory of certain AUR packages before building them? It would practically like an automatic cleanup of the corresponding cache directory before building (either explicitly or part of a system update).

The reason is that I have some packages that can be reliably built only in a clean directory, without it previously already being built before. The worst is about them is that the builds succeed, but then the applications will randomly crash.

**Comment by Morganamilo** [comment]
That already happens via -Cc to makepkg. If a package is unable to build between calls to paru it's probably an issue with the pkgbuild.

**Reply by zsolt-donca** [reply]
> If a package is unable to build between calls to paru it's probably an issue with the pkgbuild.

Well, yes, I am pretty sure that it's an issue related to the build itself. It's either an issue in the PKGBUILD or with the build of the project itself. I don't have any issue at hand now, but I've seen in the past a crash bug report closed with this exact same issue, that is, that the reporter should try a clean build.

> That already happens via -Cc to makepkg.

Can you elaborate on this? Is paru invoking makepkg with the `-Cc` argument already by default?

I see from makepkg's manpage what it does:

>       -C, --cleanbuild
>           Remove the $srcdir before building the package.
>        -c, --clean
>           Clean up leftover work files and directories after a successful build.

**Reply by Morganamilo** [reply]
Yes paru uses makepkg with those flags.

**Reply by zsolt-donca** [reply]
@Morganamilo Thank you for the clarification! I will look into more detail on exactly what causes my issues, but it doesn't seem to be on paru.

In case anyone's wondering, I was referring to about https://aur.archlinux.org/packages/wlroots-hidpi-git and https://aur.archlinux.org/packages/sway-hidpi-git .

**Reply by Morganamilo** [reply]
That's probably due to sway depending on wlroots and updating one causes ABI issues.

**Reply by zsolt-donca** [reply]
That could be it, I will further investigate.

I'm new to github discussions - how can I mark this as _Answered_?

**Reply by escape0707** [reply]
> Yes paru uses makepkg with those flags.

This is the default? I was expecting no `-Cc` as default and got unexpected `$srcdir` cleaning when using `paru`. I might need to look up the source code for where this happens.

And this can't be configured in `paru.conf`?

**Reply by escape0707** [reply]
`--cleanbuild` is used here:

https://github.com/Morganamilo/paru/blob/dbf5d8afb694a87b24318742ab2ef554b07f09bd/src/install.rs#L1425-L1430

and here:

https://github.com/Morganamilo/paru/blob/dbf5d8afb694a87b24318742ab2ef554b07f09bd/src/install.rs#L225-L230
